core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -894,7 +894,7 @@ public void updateByXml(Source source) throws IOException {
             }
 
             // try to reflect the changes by reloading
-            Object o = new XmlFile(Items.XSTREAM, out.getTemporaryFile()).unmarshalNullingOut(this);
+            Object o = new XmlFile(Items.XSTREAM, out.getTemporaryPath().toFile()).unmarshalNullingOut(this);
```

Comment:
Is this really buying us anything other than moving the `#toFile` invocation from `AtomicFileWriter` to `AbstractItem`? Either way it's the same logic as before. I don't feel strongly about keeping this change, but I also don't feel strongly about undoing it either. Up to you.

Reply:
The `getTemporaryFile()` is deprecated so I thought we can replace it with the suggested one.